packages/mantine/src/components/code-editor/CodeEditor.tsx

```diff
@@ -247,7 +245,9 @@ export const CodeEditor: FunctionComponent<CodeEditorProps> = (props) => {
                     registerThemes(monaco);
                     editor.onDidFocusEditorText(() => onFocus?.());
                     editor.onDidBlurEditorText(async () => {
-                        await editor.getAction('editor.action.formatDocument').run();
+                        if (!hasMonacoError) {
+                            await editor.getAction('editor.action.formatDocument').run();
+                        }
```

Comment:
This didn't work for me when I tried locally. It also formats in the demo when there are errors